sdk/sdk-view/src/test/java/mil/emp3/json/geoJson/GeoJsonParserTest.java

```diff
@@ -0,0 +1,154 @@
+import mil.emp3.api.interfaces.IFeature;
+import mil.emp3.api.interfaces.core.ICoreManager;
+import mil.emp3.api.interfaces.core.IEventManager;
+import mil.emp3.api.interfaces.core.IStorageManager;
+import mil.emp3.api.utils.ManagerFactory;
+import mil.emp3.core.CoreManager;
+import mil.emp3.core.EventManager;
+import mil.emp3.core.storage.StorageManager;
+import mil.emp3.core.utils.MilStdRenderer;
+import mil.emp3.mapengine.interfaces.IMilStdRenderer;
+import mil.emp3.view.R;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ManagerFactory.class, Looper.class})
+@SuppressStaticInitializationFor("mil.emp3.api.utils.ManagerFactory")
+public class GeoJsonParserTest {
+    private static final String TAG = GeoJsonParser.class.getSimpleName();
+
+    private static final String invalidJsonString = "xxxxxxxxxxxx";
+    private ClassLoader classLoader = null;
+    private static final IStorageManager storageManager = new StorageManager();
+    private static final IEventManager eventManager     = new EventManager();
+    private static final ICoreManager coreManager       = new CoreManager();
+    private static final IMilStdRenderer milStdRenderer = new MilStdRenderer();
+
+    static {
+        storageManager.setEventManager(eventManager);
+        eventManager.setStorageManager(storageManager);
+        coreManager.setStorageManager(storageManager);
+        coreManager.setEventManager(eventManager);
+        milStdRenderer.setStorageManager(storageManager);
+    }
+
+
+
+    @Before
+    public void setUp() throws Exception {
+        classLoader = GeoJsonParser.class.getClassLoader();
+        //
+        // ManagerFactory
+        //
+        PowerMockito.mockStatic(ManagerFactory.class);
+        final ManagerFactory managerFactory = PowerMockito.mock(ManagerFactory.class);
+
+        PowerMockito.when(ManagerFactory.getInstance()).thenReturn(managerFactory);
+
+        PowerMockito.when(managerFactory.getStorageManager()).thenReturn(storageManager);
+        PowerMockito.when(managerFactory.getEventManager()).thenReturn(eventManager);
+        PowerMockito.when(managerFactory.getCoreManager()).thenReturn(coreManager);
+        PowerMockito.when(managerFactory.getMilStdRenderer()).thenReturn(milStdRenderer);
+
+        //
+        // Looper
+        //
+        PowerMockito.mockStatic(Looper.class);
+        final Looper looper = PowerMockito.mock(Looper.class);
+
+        PowerMockito.when(Looper.getMainLooper()).thenReturn(looper);
+        PowerMockito.when(Looper.myLooper()).thenReturn(looper);
+    }
+
+    @After
+    public void cleanUp() throws Exception {
+
+    }
+
+    @Test
+    public void invalidJsonTest() {
+        Log.d(TAG, "Invalid json test");
+        try {
+            GeoJsonParser.parse(invalidJsonString);
+            Assert.fail("Invalid json gave no error");
+        } catch (Exception e) {
+            Log.d(TAG, "Invalid json threw exception, test passed");
+        }
```

Comment:
If you know what exception to expect then you can actually set up a test to get that exception. Some of the existing tests show how it is done OR it is there on the web.